aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3802,4 +3802,226 @@ Tensor _det_lu_based_helper_backward_helper(
   }
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ solve_triangular ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+namespace {
+void checkSameDtype(const Tensor& t1,
+                    const Tensor& t2,
+                    const char* const f_name,
+                    const char* const t1_name,
+                    const char* const t2_name) {
+  TORCH_CHECK(t1.scalar_type() == t2.scalar_type(),
+              f_name, ": Expected ", t1_name, " and ", t2_name, " to have the same dtype. ",
+              "Got ", t1_name, ".dtype = ", t1.scalar_type(),
+              " and ", t2_name, ".dtype = ", t2.scalar_type());
+}
```

Comment:
Weren't these methods introduced in one of your previous PRs to be used across all linalg functions?

Reply:
Not quite. I added them for this solver, but I intend to be using them in the rest of the solvers, and little by little (or in one large PR) use them all across `torch.linalg`, as they are much more informative than the ones we have now.

Now that I think about it though, the dtypes checks are autogen'd automatically, so this particular test is redundant. I'll delete it.